Sources/KituraWebSocket/WebSocketConnection.swift

```diff
@@ -263,12 +263,15 @@ public class WebSocketConnection {
             }
             
         case .ping:
-            if frame.payload.length < 126 {
-                sendMessage(withOpCode: .pong, payload: frame.payload.bytes, payloadLength: frame.payload.length)
-            } else {
+            guard frame.payload.length < 126 else {
                 connectionClosed(reason: .protocolError, description: "Control frames are only allowed to have payload up to and including 125 octets")
                 return
             }
+            guard frame.finalFrame else {
+                connectionClosed(reason: .protocolError, description: "Control frames cannot be fragmented")
```

Comment:
Maybe it would be a bit better to say "Control frames must not be fragmented", to match the language of the spec.